pkgs/applications/virtualization/rvvm/default.nix

```
@@ -0,0 +1,36 @@
+{ lib, stdenv, fetchFromGitHub, SDL_compat }:
+
+stdenv.mkDerivation rec {
+  pname = "rvvm";
+  version = "unstable-2023-01-25";
+
+  src = fetchFromGitHub {
+    owner = "LekKit";
+    repo = "RVVM";
+    rev = "4de27d7083db34bd074b4f056d6eb3871ccf5c10";
+    sha256 = "sha256-FjEcXfweL6FzA6iLxl9XnKaD4Fh/wZuRTJzZzHkc/B4=";
+  };
+
+  buildInputs = [ SDL_compat ];
+
+  makeFlags =
+    [ "BUILDDIR=out" "BINARY=rvvm" "USE_SDL=1" "GIT_COMMIT=${src.rev}" "all" "lib" ]
+    # work around https://github.com/NixOS/nixpkgs/issues/19098
+    ++ lib.optional (stdenv.cc.isClang && stdenv.isDarwin) "CFLAGS=-fno-lto";
+
+  installPhase = ''
+    mkdir -p $out/bin $out/lib $out/include/devices
+    cp out/rvvm $out/bin
+    cp out/librvvm.{so,a} $out/lib
+    cp src/rvvmlib.h $out/include
+    cp src/devices/*.h $out/include/devices
+  '';
```

(Upstream should really add an install target to the Makefile)
And when you override a particular phase, don't forget to run the pre and post hooks. The use of `install` is preferable as it makes sure the permissions are correct and is cleaner to read.
```suggestion
  installPhase = ''
    runHook preInstall

    install -d    $out/{bin,lib,include/devices}
    install -m755 out/rvvm           -t $out/bin
    install -m755 out/librvvm.{a,so} -t $out/lib
    install -m644 src/rvvmlib.h      -t $out/include
    install -m644 src/devices/*.h    -t $out/include/devices

    runHook postInstall
  '';
```